### PR TITLE
ENG-7685: Adding materialized view to the Txnid2-selfcheck app

### PR DESCRIPTION
**Pull request by liam700**

### CODE REVIEW

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateReplicatedMP.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 public class UpdateReplicatedMP extends ReplicatedUpdateBaseProc {
 
     public VoltTable[] run(byte cid, long rid, byte[] value, byte rollback) {
-        VoltTable[] results = doWork(r_getCIDData, r_cleanUp, r_insert, r_export, p_getAdhocData,
+        VoltTable[] results = doWork(r_getCIDData, r_cleanUp, r_insert, r_export, p_getAdhocData, r_getViewData,
```
Comment (liam700) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateReplicatedMP.java]: Should the p_getAdhocData here be r_getAdhocData?

Comment (prosegay) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateReplicatedMP.java]: seems that way, but don't change that here, use a separate pull request

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java [hunk 2]
```diff
@@ -210,7 +225,35 @@ public static void validateCIDData(VoltTable data, String callerId) {
                         " for cid " + cid + ". Got " + cntValue +
                         ", prev was: " + prevCnt);
             }
-            prevCnt = cntValue;
+            if (view != null) {
+                entries += 1;
+                max = Math.max(max, cntValue);
+                min = Math.min(min, cntValue);
+                sum += cntValue;
+                prevCnt = cntValue;
```
Comment (prosegay) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java]: why is prevCnt now conditioned by view!=null whereas before the change it wasn't?

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java [hunk 3]
```diff
@@ -210,7 +225,35 @@ public static void validateCIDData(VoltTable data, String callerId) {
...
+            }
+        }
+        if (view != null) {
+            if (view.getRowCount() != 1)
+                throw new VoltAbortException("somethin aint right");
```
Comment (prosegay) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java]: message should inform, give that another try.

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java [hunk 4]
```diff
@@ -210,7 +225,35 @@ public static void validateCIDData(VoltTable data, String callerId) {
...
+            VoltTableRow row0 = view.fetchRow(0);
+            long v_entries = row0.getLong("entries");
+            long v_max = row0.getLong("maximum");
+            long v_min = row0.getLong("minimum");
+            long v_sum = row0.getLong("summation");
+
+            if (v_entries != entries)
+                throw new VoltAbortException(
+                        "The count(*):"+v_entries+" view aggregation does not match the number of cnt entries:"+entries+" for client:"+cid);
```
Comment (prosegay) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java]: "client does not apply (any more) now that this logic is in the stored procedure...
also, say matview not view (in the future we will have views too)

File: tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java (hunk 3 above)
Comment (liam700) [inline tests/test_apps/txnid-selfcheck2/src/txnIdSelfCheck/procedures/UpdateBaseProc.java]: Oops...... forgot to fix that

**Pull request closed without merging**